pkg/syncer/cnsoperator/controller/cnsfileaccessconfig/cnsfileaccessconfig_controller.go

```diff
@@ -0,0 +1,343 @@
+			if err != nil {
+				msg := fmt.Sprintf("failed to update CnsFileAccessConfig instance: %q on namespace: %q. Error: %+v",
+					instance.Name, instance.Namespace, err)
+				recordEvent(ctx, r, instance, v1.EventTypeWarning, msg)
+				return reconcile.Result{RequeueAfter: timeout}, nil
+			}
+		}
+
+		volumeID, err := util.GetVolumeID(ctx, r.client, instance.Spec.PvcName, instance.Namespace)
+		if err != nil {
+			msg := fmt.Sprintf("Failed to get volumeID from pvcName: %q. Error: %+v", instance.Spec.PvcName, err)
+			log.Error(msg)
+			setInstanceError(ctx, r, instance, msg)
+			return reconcile.Result{RequeueAfter: timeout}, nil
+		}
+
+		// Query volume
+		log.Infof("Querying volume: %s for CnsFileAccessConfig request with name: %q on namespace: %q",
+			volumeID, instance.Name, instance.Namespace)
+		volume, err := common.QueryVolumeByID(ctx, r.volumeManager, volumeID)
+		if err != nil {
+			if err.Error() == common.ErrNotFound.Error() {
+				msg := fmt.Sprintf("CNS Volume: %s not found", volumeID)
+				log.Error(msg)
+				setInstanceError(ctx, r, instance, msg)
+				return reconcile.Result{RequeueAfter: timeout}, nil
+			}
+			log.Errorf("Failed to query CNS volume: %s with error: %+v", volumeID, err)
+			setInstanceError(ctx, r, instance, "Unable to find the volume in CNS")
+			return reconcile.Result{RequeueAfter: timeout}, nil
+		}
+
+		vSANFileBackingDetails := volume.BackingObjectDetails.(*cnstypes.CnsVsanFileShareBackingDetails)
+		accessPoints := make(map[string]string)
+		for _, kv := range vSANFileBackingDetails.AccessPoints {
+			accessPoints[kv.Key] = kv.Value
+		}
+		if len(accessPoints) == 0 {
+			log.Errorf("No access points found for volume: %q", volumeID)
+			setInstanceError(ctx, r, instance, "No access points found.")
+			return reconcile.Result{RequeueAfter: timeout}, nil
+		}
+
+		// Read the CnsFileAccessConfig instance again because the instance is already modified
+		err = r.client.Get(ctx, request.NamespacedName, instance)
+		if err != nil {
+			msg := fmt.Sprintf("Error reading the CnsFileAccessConfig with name: %q on namespace: %q. Err: %+v",
+				request.Name, request.Namespace, err)
+			// Error reading the object - requeue the request.
+			recordEvent(ctx, r, instance, v1.EventTypeWarning, msg)
+			return reconcile.Result{RequeueAfter: timeout}, nil
+		}
+
+		// Update the instance to indicate the volume registration is successful
+		msg := fmt.Sprintf("Successfully configured access points of VM: %q on the volume: %q", instance.Spec.VMName, instance.Spec.PvcName)
+		instance.Status.AccessPoints = accessPoints
+		err = setInstanceSuccess(ctx, r, instance, msg)
+		if err != nil {
+			msg := fmt.Sprintf("Failed to update CnsFileAccessConfig instance with error: %+v", err)
+			log.Error(msg)
+			setInstanceError(ctx, r, instance, msg)
+			return reconcile.Result{RequeueAfter: timeout}, nil
+		}
+		log.Info(msg)
+	}
+
+	if instance.DeletionTimestamp != nil {
+		removeFinalizerFromCRDInstance(ctx, instance)
```

Comment:
Can you add a TODO here for actions to be taken before removing finalizer?

Reply:
Added